sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryStorageStreamSource.java

```diff
@@ -219,7 +232,15 @@ private synchronized boolean readNextRecord() throws IOException {
         }
 
         fractionConsumedFromPreviousResponse = fractionConsumedFromCurrentResponse;
-        ReadRowsResponse currentResponse = responseIterator.next();
+        ReadRowsResponse currentResponse;
+        Stopwatch stopwatch = Stopwatch.createStarted();
```

Comment:
Do you think these operations can have a performance impact ? If so will it make sense to add an option to disable/enable these ?